qualtran/bloqs/mcmt/bloq_with_specialized_single_qubit_control.py

```diff
@@ -11,61 +11,58 @@
 #  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
-from typing import cast, Iterable, Optional, Protocol, runtime_checkable, Sequence, TYPE_CHECKING
+from typing import cast, Iterable, Optional, Sequence, TYPE_CHECKING
 
 import numpy as np
 
 if TYPE_CHECKING:
     from qualtran import AddControlledT, Bloq, BloqBuilder, CtrlSpec, SoquetT
 
 
-@runtime_checkable
-class BloqWithSpecializedControl(Protocol):
-    """mixin for a bloq that has a specialized single qubit controlled version."""
-
-    @property
-    def cv(self) -> Optional[int]: ...
-
-    def with_cv(self, *, cv: Optional[int]) -> Optional['Bloq']: ...
-
-    @property
-    def ctrl_reg_name(self) -> str: ...
+ControlBit = Optional[0 | 1]
 
 
 def get_ctrl_system_for_bloq_with_specialized_single_qubit_control(
-    bloq: 'BloqWithSpecializedControl', ctrl_spec: 'CtrlSpec'
+    bloq: 'Bloq',
+    ctrl_spec: 'CtrlSpec',
+    *,
+    current_control_bit: ControlBit,
+    bloq_with_ctrl: 'Bloq',
+    ctrl_reg_name: str,
+    bloq_without_ctrl: 'Bloq',
+    bloq_with_ctrl_0: Optional['Bloq'] = None,
 ) -> tuple['Bloq', 'AddControlledT']:
+    """Build the control system for a bloq with a specialized single-qubit controlled variant.
+
+    Args:
+        bloq: The current bloq of which a controlled version is requested.
+        ctrl_spec: The control specification
+        current_control_bit: The control bit of the current bloq, one of `0, 1, None`.
+        bloq_with_ctrl: The variant of this bloq with control bit `1`.
+        ctrl_reg_name: The name of the control qubit register.
+        bloq_without_ctrl: The variant of this bloq without a control.
```

Comment:
is this necessary? or more specifically: do we need all three of: `bloq`, `current_control_bit`, `bloq_without_ctrl`? maybe we can get rid of `bloq`? I just started looking at this so I haven't read the code yet; just my instinct.

edit: I think we can get rid of `bloq`

Reply:
removed `bloq`